h2o-algos/src/main/java/hex/tree/dt/CategoricalSplittingRule.java

```diff
@@ -0,0 +1,39 @@
+package hex.tree.dt;
+
+
+import java.util.Arrays;
+
+public class CategoricalSplittingRule extends AbstractSplittingRule {
+
+  public CategoricalSplittingRule(int featureIndex, boolean[] mask, double criterionValue) {
+    _featureIndex = featureIndex;
+    // categories for the left split - bitmask
+    _mask = mask;
+    _criterionValue = criterionValue;
+  }
+
+  public CategoricalSplittingRule(boolean[] mask) {
+    _featureIndex = -1; // tmp value, non initialized yet
```

Comment:
What do you think about moving this initialization to the parent class as default values?  

```suggestion
    
public abstract class AbstractSplittingRule extends Iced<AbstractSplittingRule> {

  protected int _featureIndex = -1;
  protected double _criterionValue = -1;
```